integration-tests/webhook/webhook_test.go

```diff
@@ -0,0 +1,163 @@
+package webhook
+
+import (
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openebs/maya/integration-tests/artifacts"
+	k8s "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	cvr "github.com/openebs/maya/pkg/cstorvolumereplica/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// stsYaml holds the yaml spec
+	// for statefulset application
+	clonePVCYaml artifacts.Artifact = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-snap-vol-claim
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: fastfurious
+spec:
+  storageClassName: openebs-snapshot-promoter
+  accessModes: [ "ReadWriteOnce" ]
+  resources:
+    requests:
+      storage: 2G`
+)
+
+var _ = Describe("AdmissionWebhook", func() {
+	BeforeEach(func() {
+		// Extracting storageclass artifacts unstructured
+		_, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(artifacts.SingleReplicaSC),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting PVC artifacts unstructured
+		PVCUnst, err := artifacts.GetArtifactUnstructured(artifacts.Artifact(artifacts.CStorPVCArtifacts))
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting statefulset application namespace
+		PVCNamespace := PVCUnst.GetNamespace()
+
+		// Webhook stuffs
+		client, err := k8s.Clientset().Get()
```

Comment:
Please write idiomatic code for CRUD operations against validating webhook.
The code should be available at:
`pkg/kubernetes/webhook/validate/v1alpha1/validate.go` -- will have creational patterns
`pkg/kubernetes/webhook/validate/v1alpha1/kubernetes.go` -- will have k8s client based CRUD ops

When the case for mutating webhook comes up, let us package the code at:
`pkg/kubernetes/webhook/mutate/v1alpha/mutate.go` -- will have creational patterns
`pkg/kubernetes/webhook/mutate/v1alpha/kubernetes.go` -- will have k8s client based CRUD ops

Reply:
Done